telosnetwork/telos-wallet
# #645 | IPFS NFTs

# Fixes #645 

Includes work by @Viterbo from https://github.com/telosnetwork/telos-wallet/pull/667

## Description
This PR enables viewing of IPFS NFTs. It also fixes a few issues:
- video NFTs now correctly show the hover preview when in icon mode (such as list view - NFT inventory page)
- very tall video NFTs no longer break the NFT detail page layout
- NFT detail cards (on the NFT detail page, such as "Owner" and "Description") no longer grow in height when used with very tall NFTs
- you can now access the NFT detail page when not logged in (fixed by @Viterbo - underlying issue was that the contract manager relied on a Signer, but now uses a Provider as a fallback)

## Test scenarios
- go to https://deploy-preview-665--wallet-develop-mainnet.netlify.app/
- log in
- go to an ERC721 NFT detail page, such as https://deploy-preview-665--wallet-develop-mainnet.netlify.app/evm/collectible-details?contract=0xb0DdE512a8f996412312491715c5936cee35D9F4&id=1007&tab=attributes
    - the layout should look good
- go to an ERC1155 NFT detail page, such as https://deploy-preview-665--wallet-develop-mainnet.netlify.app/evm/collectible-details?contract=0xB63d03F03A7754a7E6c9A38C312fe21e0b966a8E&id=1&tab=attributes
    - the layout should look good
- go to an IPFS NFT detail page, such as https://deploy-preview-665--wallet-develop-mainnet.netlify.app/evm/collectible-details?contract=0xcfbed196e6e53e076BC06bAc40462E7A6B594187&id=1&tab=attributes
    - you should see the NFT 


Hover working for video NFTs:

https://github.com/telosnetwork/telos-wallet/assets/11837820/841282a6-da35-436e-90ac-dd399c10e7c2



## Checklist:
